CMakePresets.json

```diff
@@ -0,0 +1,28 @@
+{
+  "version": 1,
+  "cmakeMinimumRequired": {
+    "major": 3,
+    "minor": 19,
+    "patch": 0
+  },
+  "configurePresets": [
+    {
+      "name": "base",
+      "hidden": true,
+      "binaryDir": "${sourceDir}/build",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "Release"
+      }
+    },
+    {
```

Comment:
why not use ninja by default, and let the makefile one be the distinct one? additionally, you can add configurations for windows here both for MSVC and Clang, take a look at https://github.com/neovim/neovim/pull/19128/files (would love more testers for the PR btw!)

Reply:
That's using a more advanced version of presets. Your minimum CMake version is 3.2, so I didn't want to get wild with the configure/build presets 😅

Reply:
As for ninja by default, that's fine too. As long as I *personally* can configure + build in one or two steps to keep my packer.nvim post-pull step minimal, I'll be happy 😛